<launch>
  <arg name="child_frame_id" default="lslidar"/>
  <arg name="nodelet_manager_name" value="velodyne16_nodelet_manager" />
  <arg name="device_ip" value="192.168.1.200"/>
  <arg name="firing_port" value="2368"/>
  <arg name="topic_packet" default="/apollo/sensor/velodyne16/LslidarPacket"/>
  <arg name="topic_pointcloud" default="/apollo/sensor/velodyne16/PointCloud2"/>
  <arg name="topic_scan_unified" default="/apollo/sensor/velodyne16/VelodyneScanUnified"/>
  <arg name="topic_pc_compensator" default="/apollo/sensor/velodyne16/compensator/PointCloud2"/>
  
    <!-- nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" output="screen"
        args="manager" />

    <!-- nodelet driver for lslidar_c32 -->
  <node pkg="nodelet" type="nodelet"
    name="lslidar_c32_driver_nodelet"
    args="load lslidar_c32_driver/LslidarC32DriverNodelet
    $(arg nodelet_manager_name)" >
    <param name="frame_id" value="lslidar"/>
    <param name="device_ip" value="$(arg device_ip)"/>
    <param name="device_port" value="$(arg firing_port)"/>
    <remap from="lslidar_packet" to="$(arg topic_packet)"/>
  </node>
  

  <node pkg="nodelet" type="nodelet" name="lslidar_c32_decoder_nodelet"
    args="load lslidar_c32_decoder/LslidarC32DecoderNodelet
    $(arg nodelet_manager_name)"
    output="screen">
    <param name="child_frame_id" value="$(arg child_frame_id)"/>
    <param name="min_range" value="0.3"/>
    <param name="max_range" value="100.0"/>
    <param name="frequency" value="10.0"/>
    <param name="publish_point_cloud" value="true"/>
    <param name="publish_channels" value="false"/>
    <remap from="lslidar_packet" to="$(arg topic_packet)"/>
    <remap from="lslidar_point_cloud" to="$(arg topic_pointcloud)"/>
  </node>
  
 <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find lslidar_c32_decoder)/launch/lslidar_c32.rviz" output="screen"/>
-->
</launch>
